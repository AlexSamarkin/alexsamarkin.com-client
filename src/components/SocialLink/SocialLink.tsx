import React from 'react';

export interface SocialLinkProps {
    link: string;
    icon: string;
}

export const SocialLink: React.FC<SocialLinkProps> = ({ link, icon }) => {
    const className = `font-icon ${icon}`;
    return (
        <a target="_blank" rel="noreferrer" className="social__link" href={link}>
            <i className={className} />
        </a>
    );
};

export default SocialLink;
