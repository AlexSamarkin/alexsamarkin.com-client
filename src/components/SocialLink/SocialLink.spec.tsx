import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SocialLink from './SocialLink';

Enzyme.configure({ adapter: new Adapter() });

describe('SocialLink component', () => {
    test('renders', () => {
        const wrapper = shallow(<SocialLink link={'https://social.link'} icon={'icon.svg'} />);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});
