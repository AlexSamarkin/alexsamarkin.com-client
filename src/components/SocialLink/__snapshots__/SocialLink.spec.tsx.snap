// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink component renders 1`] = `ShallowWrapper {}`;
